From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Wed, 3 Jul 2024 14:12:17 +0200
Subject: qt: hardcode the name of the shortcut for the AMD VQ Enhancer

Otherwise the name is the same of the DLL which doesn't correspond to
the shortcut we need to use.

Fixes #28691
---
 modules/gui/qt/components/preferences_widgets.cpp | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/modules/gui/qt/components/preferences_widgets.cpp b/modules/gui/qt/components/preferences_widgets.cpp
index 64121b6..b634d53 100644
--- a/modules/gui/qt/components/preferences_widgets.cpp
+++ b/modules/gui/qt/components/preferences_widgets.cpp
@@ -632,9 +632,15 @@ ModuleListConfigControl::~ModuleListConfigControl()
 void ModuleListConfigControl::checkbox_lists( module_t *p_parser )
 {
     const char *help = module_get_help( p_parser );
-    checkbox_lists( qtr( module_GetLongName( p_parser ) ),
+    const char *module_name = module_GetLongName( p_parser );
+    const char *module_shortcut = module_get_object( p_parser );
+
+    if ( !strcmp(module_name, "AMD VQ Enhancer"))
+        module_shortcut = "amf_vqenhancer";
+
+    checkbox_lists( qtr( module_name ),
                     help != NULL ? qtr( help ): "",
-                    module_get_object( p_parser ) );
+                    module_shortcut );
 }
 
 void ModuleListConfigControl::checkbox_lists( QString label, QString help, const char* psz_module )
